include/counter.h

```diff
@@ -60,9 +59,7 @@ struct counter_alarm_cfg {
 		      *   counter implementation may count asynchronous events.
 		      */
 
-	u8_t channel_id; /*!< Channel ID. Number of channels is driver
-			  * dependent.
-			  */
+	void *user_data; /*!< User data returned in callback.*/
```

Comment:
This is somehow unrelated to this PR but still needs to be addressed. I've generated locally Zephyr's documentation with `ninja htmldocs` and as expected the `/*!< User data returned in callback.*/` placed after the field name was ignored by doxygen. A working example of documenting structure fields can be looked up among others in i2s.h API searching for `struct i2s_config`.

Reply:
fixed